extensions/amp-lightbox-gallery/0.1/test/validator-amp-lightbox-gallery.out

```diff
@@ -0,0 +1,78 @@
+|    limitations under the license.
+|  -->
+|  <!--
+|    Test Description:
+|    Tests for the amp-lightbox-gallery tag. See inline comments.
+|  -->
+|  <!doctype html>
+|  <html ⚡>
+|  <head>
+|    <meta charset="utf-8">
+|    <link rel="canonical" href="./regular-html-version.html">
+|    <meta name="viewport" content="width=device-width,minimum-scale=1">
+|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+|    <script async src="https://cdn.ampproject.org/v0.js"></script>
+|    <script async custom-element="amp-lightbox-gallery" src="https://cdn.ampproject.org/v0/amp-lightbox-gallery-0.1.js"></script>
+|    <script async custom-element="amp-youtube" src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"></script>
+|    <script async custom-element="amp-carousel" src="https://cdn.ampproject.org/v0/amp-carousel-0.1.js"></script>
+|    <script async custom-element="amp-video" src="https://cdn.ampproject.org/v0/amp-video-0.1.js"></script>
+|  
+|  </head>
+|  <body>
+|    <!-- Valid -->
+|    <amp-img src="/awesome.png" width="300" height="300" lightbox></amp-img>
+|    <amp-img src="/awesome.png" width="300" height="300" lightbox="a"></amp-img>
+|    <amp-img src="/awesome.png" width="300" height="300" lightbox lightbox-thumbnail-id="a"></amp-img>
+|  
+|    <!-- Invalid: missing value for lightbox-thumbnail-id -->
+|    <amp-img src="/awesome.png" width="300" height="300" lightbox lightbox-thumbnail-id></amp-img>
+>>   ^~~~~~~~~
+amp-lightbox-gallery/0.1/test/validator-amp-lightbox-gallery.html:41:2 The attribute 'lightbox-thumbnail-id' in tag 'amp-img' is set to the invalid value ''. (see https://www.ampproject.org/docs/reference/components/amp-img) [AMP_TAG_PROBLEM]
+|    <!-- Invalid: missing lightbox attribute for element with lightbox-thumbnail-id -->
+|    <amp-img src="/awesome.png" width="300" height="300" lightbox-thumbnail-id="a"></amp-img>
+>>   ^~~~~~~~~
+amp-lightbox-gallery/0.1/test/validator-amp-lightbox-gallery.html:43:2 The attribute 'lightbox' in tag 'amp-img' is missing or incorrect, but required by attribute 'lightbox-thumbnail-id'. (see https://www.ampproject.org/docs/reference/components/amp-img) [DISALLOWED_HTML]
+|    <!-- Invalid: cannot have lightbox-exclude on an item that isn't in a carousel -->
```

Comment:
Note that this isn't being marked as invalid. It seems like you want `lightbox-exclude` to only be an attribute allowed on `amp-carousel` children. The only way to enforce that currently would be to add it as an allowed attribute for all possible children tags of `amp-carousel`.

Reply:
Yup. Let me remove this test case for now. What I intend to do is you can only have `lightbox-exclude` on children of `<amp-carousel>` that have the `lightbox` attribute. I think the proper way to do this is to add a reference point in `<amp-carousel>` refering to `<amp-carousel>`s with the lightbox attribute, and then to add said reference point as a mandatory parent in `lightbox-exclude` under the main validator `.protoascii` file.

Reply:
Okay, it seems like we can't add `mandatory_parent` or `mandatory_ancestor` attributes to to `attr`. 

I tried adding a reference point to `<amp-carousel>` that applies to its children. However, this conflicts with the `<amp-selector>` reference points, which cover all children of `<amp-selectors>`.